File: source_funcs/go_transcoder/K_TH_DISTINCT_OR_NON_REPEATING_ELEMENT_IN_AN_ARRAY/K_TH_DISTINCT_OR_NON_REPEATING_ELEMENT_IN_AN_ARRAY.go
package main

func f_gold(arr []int, n int, k int) int {
	var dist_count int = 0
	for i := int(0); i < n; i++ {
		var j int
		for j = 0; j < n; j++ {
			if i != j && arr[j] == arr[i] {
				break
			}
		}
		if j == n {
			dist_count++
		}
		if dist_count == k {
			return arr[i]
		}
	}
	return -1
}

File: rust_funcs/cpp_transcoder/K_TH_DISTINCT_OR_NON_REPEATING_ELEMENT_IN_AN_ARRAY/K_TH_DISTINCT_OR_NON_REPEATING_ELEMENT_IN_AN_ARRAY.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}


fn f_gold(arr: [f32;2], n: f32, k: f32) -> i32 {
    let mut dist_count = 0;
    for i in 0..n as usize {
        let mut j = 0;
        while j < n as usize && arr[j] != arr[i] {j += 1;}
        if j == n as usize {dist_count += 1;}
        if dist_count == k as usize {return arr[i];}
    }
    -1.0  
}
